cardano-sl.cabal

```diff
@@ -512,10 +512,13 @@ library
                        -fno-warn-orphans
                        -O2
 
-  -- linker speed up for linux
+  -- linker speed up for linux and macOS
   if os(linux)
     ghc-options:       -optl-fuse-ld=gold
     ld-options:        -fuse-ld=gold
+  if os(darwin)
+    ghc-options:       "-pgmc clang" "-pgma clang" "-pgml clang" "-optl-fuse-ld=lld"
+    ld-options:        -fuse-ld=lld
```

Comment:
I don't think lld is in a state where can be used on Mach-O, see my comment.
Also, it's not necessarily faster than gold. In many tests we run it goes from parity to 2x faster (in particular scenarios, e.g. when linking with debuginfo or with some options (e.g. `-gdb-index`). It's a safe claim to say that's generally at least as fast as gold.